classes/sample_set.php

```diff
@@ -30,18 +30,53 @@ class sample_set {
     public $samples = [];
     public $starttime;
 
+    public $limit;
 
-    public function __construct($name, $starttime) {
+    public $filterrate = 1;
+    public $current = 0;
+
+    /**
+     * Constructs the sample set.
+     *
+     * @param string $name
+     * @param float $starttime
+     * @param float|null $limit
+     */
+    public function __construct(string $name, float $starttime, int $limit = null) {
         $this->name = $name;
         $this->starttime = $starttime;
+        $this->limit = $limit;
     }
 
     /**
-     * Add a sample the sample store.
+     * Add a sample to the sample store, applying any filters.
      *
      * @param array $sample
      */
     public function add_sample(\ExcimerLogEntry $sample): void {
-        $this->samples[] = $sample;
+        if (count($this->samples) === $this->limit) {
+            $this->apply_doubling();
+        }
+        $this->current += 1;
+        if ($this->current == $this->filterrate) {
```

Comment:
Should use `===` instead of `==` because it seems it should always be an `int` comparison.

Reply:
done